migrations/V177__Add_Data_Exchange_Approach_Marked_Complete_Notification.sql

```diff
@@ -0,0 +1,15 @@
+CREATE TYPE DATA_EXCHANGE_APPROACH_MARKED_COMPLETE_NOTIFICATION_TYPE as enum (
+  'ALL_MODELS',
+  'FOLLOWED_MODELS',
+  'MY_MODELS'
+  );
+
+
+ALTER TYPE ACTIVITY_TYPE
+ADD VALUE 'DATA_EXCHANGE_APPROACH_MARKED_COMPLETE' AFTER 'NEW_MODEL_PLAN';
+
+ALTER TABLE user_notification_preferences
+  ADD COLUMN data_exchange_approach_marked_complete USER_NOTIFICATION_PREFERENCE_FLAG[] DEFAULT '{}'::USER_NOTIFICATION_PREFERENCE_FLAG[],
+  ADD COLUMN data_exchange_approach_marked_complete_notification_type DATES_CHANGED_NOTIFICATION_TYPE DEFAULT NULL;

```

Comment:
Maybe there should be a constraint on these columns? Something to ensure that if a user has non empty preferences for `data_exchange_approach_marked_complete` that `data_exchange_approach_marked_complete_notification_type` is not null? Otherwise, it probably needs to be handled in the app code.

This could be fine as is, I think effectively they just wont get any notifications, because the SQL will not return their preferencs if the type column is not filled out. It might be beneficial to enforce though so we don't have unexpected behavior.

Reply:
Discussed with @ClayBenson94, we are going to keep it homogenous with the dates change feature for now as a matter of consistency, then do a tech debt pass to update both solutions to default notification types to 'ALL_MODELS' and NOT NULL constraint.